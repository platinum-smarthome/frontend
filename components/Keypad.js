import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native'
import { devicePinUpdate } from '../store/data/data.actions'

class Keypad extends Component {
  render() {
    return (
      <View style={ styles.keybox }>
          <Text style={ styles.keypress } onPress={() => this.props.press('1')}>1</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('2')}>2</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('3')}>3</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('4')}>4</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('5')}>5</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('6')}>6</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('7')}>7</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('8')}>8</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('9')}>9</Text>
          <Text style={ styles.keypress } >&nbsp;</Text>
          <Text style={ styles.keypress } onPress={() => this.props.press('0')}>0</Text>
          <Text style={ styles.keypress } onPress={() => this.props.remove() }>&#8592;</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  keybox: {
    justifyContent: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  keypress: {
    flex: 1,
    color: '#fff',
    fontWeight: '200',
    fontSize: 30,
    paddingVertical: 10,
    paddingHorizontal: 36,
    // marginHorizontal: 16
    marginHorizontal: 24
  }
});

export default Keypad