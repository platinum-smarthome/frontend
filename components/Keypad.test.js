import React, { Component } from 'react'
import enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import Keypad from './Keypad'
import { Text, View } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper
let mockFunction
let wrap

beforeEach(() => {
  wrapper = shallow(<Keypad />)
  mockFunction = jest.fn()
  mockFunctionRemove = jest.fn()
  wrap = shallow(<Keypad press={mockFunction} remove={mockFunctionRemove}/>)
})

describe('<Keypad/>', () => {
  it('should render <Keypad/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<Keypad/> child is rendered', () => {
  it('Should render <View/>', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('View').length).toBe(1)  
  })
  it('Should render <Text/>', () => {
    expect(wrapper.find(<Text/>)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(12)  
  })
})

describe('<Text/> could be pressed', () => {
  it('Button 1 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(0).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 2 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(1).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 3 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(2).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 4 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(3).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 5 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(4).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 6 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(5).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 7 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(6).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 8 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(7).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 9 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(8).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
  it('Button 0 should be able to be pressed', () => {
    expect(mockFunction).not.toHaveBeenCalled()
    wrap.find('Text').at(10).props().onPress()
    expect(mockFunction).toHaveBeenCalled()
  })
})

describe('<Text/> should be able to remove pin input', () => {
  it('Backspace button should be able to be pressed', () => {
    expect(mockFunctionRemove).not.toHaveBeenCalled()
    wrap.find('Text').at(11).props().onPress()
    expect(mockFunctionRemove).toHaveBeenCalled()
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<Keypad/>)
    expect(tree).toMatchSnapshot()
  })
})