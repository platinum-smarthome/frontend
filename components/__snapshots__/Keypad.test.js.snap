// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "marginBottom": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    1
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    2
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    3
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    5
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    6
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    7
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    8
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    9
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
     
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "200",
        "marginHorizontal": 6,
        "paddingHorizontal": "13%",
        "paddingVertical": 10,
      }
    }
  >
    ←
  </Text>
</View>
`;
